.github/workflows/pr.yaml

```diff
@@ -0,0 +1,50 @@
+name: CI for PRs
+
+on:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+    branches:
+      - main
+
+jobs:
+  ci_checks:
+    name: CI Checks
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        package:
+          - packages/yx_scope
+          - packages/yx_scope_flutter
+          - packages/yx_scope_linter
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: '3.27.0'
+
+      - name: Install dependencies
+        run: |
+          cd ${{ matrix.package }}
```

Comment:
can be replaced by default `working-directory` for a job

```yaml
    defaults:
      run:
        working-directory: ./scripts
```


https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-setting-default-run-step-options-for-a-job

this could also fix logs, only the first line of the `run` block is currently visible

<img width="1128" alt="image" src="https://github.com/user-attachments/assets/275a64d2-0c63-46db-9754-628862f14099" />